### ISSUE TITLE
Create SQL Parser

### ISSUE THREAD

**Issue by matthias-Q** [matthias-Q/rusqlls#1]
We need to be able to parse SQL statements.

[Chumsky](https://github.com/zesterer/chumsky) seems to be a good library, as it allows for very good error handling. 

Not sure how to structure the process at the moment so I will just go ahead and create a list of
statement that need to be parse-able:

- [ ] SELECT statements
- [ ] CREATE statements
- [ ] UPDATE statements
- [ ] DELETE statements
- [ ] DROP (table, index)
- [ ] Function calls
- [ ] Comments
- [ ] ALTER (table, index)

**Comment by atcol** [comment]
Hello I saw your post on reddit and thought I'd see if I can help.

Have you thought about using https://github.com/sqlparser-rs/sqlparser-rs?

**Comment by matthias-Q** [comment]
Yes, we did take a brief look into it. I would like to use it, in order to re-implement a parser, but it looks like it is not "error resilient"
